### PR TITLE
Improve RestPlatformChannel publish

### PR DESCRIPTION
**Pull request by zoechi**
replaces #36

### CODE REVIEW

File: lib/src/method_call_handler.dart [hunk 1]
```diff
@@ -3,26 +3,23 @@ import 'package:ably_flutter_plugin/src/generated/platformconstants.dart';
...
     ably.TokenParams tokenParams = message.message as ably.TokenParams;
     ably.Rest rest = ably.restInstances[message.handle];
-    Object callbackResponse = await rest.options.authCallback(tokenParams);
-    Future.delayed(Duration.zero, (){
+    await rest.options.authCallback(tokenParams);
+    Future.delayed(Duration.zero, () {
       rest.channels.all.forEach((c) => c.authUpdateComplete());
     });
-    return callbackResponse;
+    return;
```
Comment (tiholic) [inline lib/src/method_call_handler.dart:23]: @zoechi the returned `callbackResponse` will be returned back to platform. So, we'd need this return statement to be as is.

File: lib/src/impl/rest/channels.dart [hunk 2]
```diff
@@ -1,14 +1,16 @@
...
 import 'package:ably_flutter_plugin/src/impl/rest/rest.dart';
 import 'package:ably_flutter_plugin/src/spec/spec.dart' as spec;
 import 'package:flutter/services.dart';
 
 import '../platform_object.dart';
 
-
-class RestPlatformChannel extends PlatformObject implements spec.RestChannel{
+class RestPlatformChannel extends PlatformObject implements spec.RestChannel {
+  static const String clientConfiguredAuthenticationProviderRequestFailed =
```
Comment (tiholic) [inline lib/src/impl/rest/channels.dart:12]: @zoechi A new commit (e9e0939) has been cherry-picked from another latest branch into `feature/realtime-authcallback`.

You may rebase and use that constant

**Review by tiholic — CHANGES_REQUESTED** [review]

File: lib/src/method_call_handler.dart (hunk 1 above)
Comment (zoechi) [reply lib/src/method_call_handler.dart:23]: The function should have a proper return type then.
The type annotation in the original code is `Object` (line 21), but `AuthCallback` is defined to return `dynamic`. 

I think I suggested to not use `dynamic` at all already (except a few exceptions where it's unavoidable).
I'll leave `dynamic` here anyway for consistency.

If we actually want to to consistently not use `dynamic`, we should create an issue for this and make a PR where we get rid of it everywhere.

File: lib/src/impl/rest/channels.dart (hunk 2 above)
Comment (zoechi) [reply lib/src/impl/rest/channels.dart:12]: `PlatformException` has `final String code`, but `ErrorCodes` has `static const int authCallbackFailure`
Does this type inconsistency `String`/`int` make any sense?

File: lib/src/method_call_handler.dart (hunk 1 above)
Comment (tiholic) [reply lib/src/method_call_handler.dart]: Sure, let's get rid of everything `dynamic`!

File: lib/src/impl/rest/channels.dart (hunk 2 above)
Comment (tiholic) [reply lib/src/impl/rest/channels.dart]: This discussion is continued either https://github.com/ably/ably-flutter/pull/38#discussion_r484375487 or https://github.com/ably/ably-flutter/pull/38#discussion_r484379978

**Review by tiholic — APPROVED** [review]

**Pull request merged**